### ISSUE TITLE
App crash when pressing on Albums tab

### ISSUE THREAD

**Issue by sonalder-darlene** [IacobIonut01/Gallery#39]
App version : 1.0.2
OS : GrapheneOS (stable)
Android version : 13

Every time I press on the Albums tab the app crash. Here are the logs I got :

```
type: crash
osVersion: google/bluejay/bluejay:13/TQ2A.230405.003.E1/2023041100:user/release-keys
package: com.dot.gallery:10216
process: com.dot.gallery
processUptime: 12111 + 568 ms

java.lang.IndexOutOfBoundsException: Empty list doesn't contain element at index 0.
	at a9.r.get(Unknown Source:21)
	at w5.c.j(Unknown Source:63)
	at w5.j.a(Unknown Source:14)
	at i7.e.L(Unknown Source:169)
	at q0.b.b(Unknown Source:50)
	at q0.b.L(Unknown Source:8)
	at o.l0.a(Unknown Source:130)
	at o.l0.S(Unknown Source:88)
	at q0.b.S(Unknown Source:58)
	at o5.c.d(Unknown Source:183)
	at s9.y.i(Unknown Source:184)
	at v.q.L(Unknown Source:376)
	at q0.b.b(Unknown Source:50)
	at q0.b.L(Unknown Source:8)
	at w.d.j(Unknown Source:79)
	at v.r.j(Unknown Source:61)
	at w.a.j(Unknown Source:68)
	at v.s.j(Unknown Source:12)
	at o.l0.a(Unknown Source:89)
	at o.l0.S(Unknown Source:76)
	at q0.b.S(Unknown Source:58)
	at o5.c.d(Unknown Source:183)
	at r0.i.a(Unknown Source:114)
	at w.f0.a(Unknown Source:32)
	at q.a0.b(Unknown Source:1256)
	at q.a0.S(Unknown Source:320)
	at q0.b.S(Unknown Source:58)
	at q.a0.b(Unknown Source:732)
	at q.a0.S(Unknown Source:70)
	at q0.b.S(Unknown Source:58)
	at o5.c.r0(Unknown Source:19)
	at v.t.a(Unknown Source:366)
	at v.t.n(Unknown Source:45)
	at o5.c.B0(Unknown Source:42)
	at j0.z.p(Unknown Source:128)
	at j0.z.j(Unknown Source:13)
	at j0.e0.m(Unknown Source:17)
	at j0.a2.a(Unknown Source:38)
	at j0.k.a(Unknown Source:9)
	at j0.k.a(Unknown Source:9)
	at j0.k.a(Unknown Source:9)
	at j0.e0.e(Unknown Source:10)
	at l1.z.c(Unknown Source:101)
	at l1.v.a(Unknown Source:134)
	at w.q.a(Unknown Source:35)
	at v.s0.a(Unknown Source:13)
	at v.t0.b(Unknown Source:51)
	at v.d0.S(Unknown Source:776)
	at q.a0.S(Unknown Source:305)
	at l1.x.a(Unknown Source:47)
	at n1.q.c(Unknown Source:41)
	at p.g.e(Unknown Source:60)
	at p.g.R(Unknown Source:432)
	at l1.t.d(Unknown Source:12)
	at n1.x.c(Unknown Source:10)
	at p.g.e(Unknown Source:19)
	at p.g.R(Unknown Source:445)
	at l1.t.d(Unknown Source:12)
	at n1.x.c(Unknown Source:10)
	at z0.e0.d(Unknown Source:5)
	at n1.x.c(Unknown Source:10)
	at t.c0.d(Unknown Source:95)
	at n1.e.d(Unknown Source:14)
	at n1.x.c(Unknown Source:10)
	at t.s0.d(Unknown Source:37)
	at n1.e.d(Unknown Source:14)
	at n1.x.c(Unknown Source:10)
	at n1.g0.a(Unknown Source:41)
	at n1.g0.n(Unknown Source:16)
	at i0.a.d(Unknown Source:63)
	at p.s0.a(Unknown Source:121)
	at p.s0.n(Unknown Source:136)
	at o5.c.B0(Unknown Source:42)
	at s0.z.c(Unknown Source:117)
	at n1.i1.a(Unknown Source:12)
	at n1.j0.r0(Unknown Source:116)
	at n1.d0.M(Unknown Source:16)
	at n1.o0.c(Unknown Source:24)
	at n1.o0.j(Unknown Source:124)
	at n1.o0.f(Unknown Source:70)
	at androidx.compose.ui.platform.AndroidComposeView.m(Unknown Source:13)
	at n1.f1.a(Unknown Source:3)
	at androidx.activity.b.run(Unknown Source:284)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7891)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ExecInit.main(ExecInit.java:49)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:355)

```

**Comment by sonalder-darlene** [comment]
I achieved to see the albums tab for a second or two. But when I tried again it crashed.

**Comment by IacobIonut01** [comment]
Can you send a list of each path you have an Album on?

Like:
/storage/emulated/0/DCIM/Camera
/../DCIM/Messenger
/../Pictures/..
/../MyAlbum

Also if you have any media in the root of the storage like (/storage/emulated/0/*.jpg,png etc)

> For privacy reasons the list can be sent directly to me via email: paulionut2003@gmail.com

**Comment by sonalder-darlene** [comment]
I don't know what changes but sometimes it's working well and sometimes it crashs.

I have album mainly in the Pictures, Downloads and DCIM folders. Can't see where I could have put the others.

Also GMail and privacy don't match well ;)

**Comment by IacobIonut01** [comment]
> I don't know what changes but sometimes it's working well and sometimes it crashs.
> 
> I have album mainly in the Pictures, Downloads and DCIM folders. Can't see where I could have put the others.
> 
> Also GMail and privacy don't match well ;)

Do you have any special names for any of the Albums (emojis are fine)?

**Comment by IacobIonut01** [comment]
Try this apk and try to log again the crash again
[app-staging.zip](https://github.com/IacobIonut01/Gallery/files/11302108/app-staging.zip)

**Comment by sonalder-darlene** [comment]
I have some folder with " - " or " _ " symbol and one with " ' " but beside that it's only letters and numbers.
Will try the APK

**Comment by IacobIonut01** [comment]
> Album tab crashed for me too.
> 
> It only happens when I switch to the Album tab right after launching the app. If I wait a few seconds before switching, nothing happens.

Please check the apk from above and try to log the issue

**Comment by IacobIonut01** [comment]
Fixed in 1.1 beta release
